nubison_model/Model.py

```diff
@@ -18,20 +18,52 @@ def load_model(self) -> None: ...
     def infer(self, input: any) -> any: ...
 
 
+def _is_shareable(package: str) -> bool:
+    # Nested requirements, constraints files, local packages, and comments are not supported
+    if package.startswith(("-r", "-c", "-e .", "-e /", "/", ".", "#")):
+        return False
+    # Check if the package is a local package
+    # eg. git+file:///path/to/repo.git, file:///path/to/repo, -e file:///
+    if "file:" in package:
+        return False
+
+    return True
```

Comment:
Since both conditions look for patterns indicating non-shareable packages and return the same False result, it seems like it would be cleaner to combine them.

For example:

```
def _is_shareable(package: str) -> bool:

    non_shareable_patterns = ("-r", "-c", "-e .", "-e /", "/", ".", "#", "file:")
    
    is_non_shareable = any(package.startswith(pattern) for pattern in non_shareable_patterns)

    return not is_non_shareable
```
Combining them this way improves readability. Is there a reason the two patterns need to be separated, or do they serve different purposes?

Reply:
Yes, the added condition doesn’t look very ideal, does it?
The first condition checks for lines that start with -r, -c, etc.
The second condition checks for lines that contain file: in the middle, like in these examples:
`file:///some_path`
`git+file:///some_path`
`package_name @ svn+file:///another_path`

This was the best I could come up with. Do you have any suggestions to make it better?